jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt

```diff
@@ -105,8 +108,28 @@ class CawsConnectionProvider : GatewayConnectionProvider {
 
         val ssoSettings = connectionParams.ssoSettings ?: SsoSettings(SONO_URL, SONO_REGION)
 
+        val currentConnection = ToolkitConnectionManager.getInstance(null).activeConnectionForFeature(CodeCatalystConnection.getInstance())
+            as AwsBearerTokenConnection?
+        if (ssoSettings.startUrl != currentConnection?.startUrl) {
+            val ans = Messages.showOkCancelDialog(
+                message("gateway.auth.different.account.required"),
+                message("caws.login"),
+                message("gateway.auth.different.account.sign.in"),
+                message("general.cancel"),
+                Messages.getErrorIcon(),
+                null
+            )
+            if (ans == 0) {
+                if (currentConnection != null) {
+                    logoutFromSsoConnection(project = null, currentConnection)
+                }
+                requestCredentialsForCodeCatalyst(project = null)
```

Comment:
i think you also need to handle the cancel on this request?

Reply:
doesn't this pop up a free-form dialog? shouldn't we just log them in directly to the account they need?